pkg/client/events/replay_client.go

```diff
@@ -191,7 +219,20 @@ func (rClient *replayClient[T, R]) retryPublishEventsFactory(ctx context.Context
 			rClient.newMapEventsBytesToTFn(errCh),
 		)
 
-		// Initially set latestObsvbls and update if after retrying on error.
+		// Subscribe to the eventBzObs and block until the channel closes.
+		// Then pass this as an error to force the retry.OnError to resubscribe.
+		go func() {
+			eventsBzSub := eventsBz.Subscribe(ctx)
+			for range eventsBzSub.Ch() {
+				// Wait for the channel to close.
+				continue
+			}
+			typedObs.UnsubscribeAll()
```

Comment:
Would you mind adding a comment to remind us why we need to do this?

Reply:
IIRC it's because `typedObs` is a derivative of `eventsBzObs` + the fact that calling `UnsubscribeAll()` on an observable has no effect observables derived from it (i.e. their observers would still be subscribed).